src/main/java/org/wildfly/security/auth/provider/FileSystemSecurityRealm.java

```diff
@@ -423,7 +427,7 @@ public void setCredentials(final List<Object> credentials) throws RealmUnavailab
 
         public void setAttributes(final Attributes attributes) throws RealmUnavailableException {
             Assert.checkNotNullParam("attributes", attributes);
-            final LoadedIdentity loadedIdentity = loadIdentity(false, true);
+            final LoadedIdentity loadedIdentity = loadIdentity(false, false);
```

Comment:
Why are you disable `skipAttributes` here? You dont use `loadedIdentity.getAttributes()` in this method.